System.Security.Cryptography/Aes.cs

```diff
@@ -27,17 +28,39 @@ public class Aes
         /// <value>The secret key for the symmetric algorithm.</value>
         public byte[] Key { get => _key; set => _key = value; }
 
+        /// <summary>
+        /// Gets or sets the initialization vector for the symmetric algorithm.
+        /// </summary>
+        /// <value>The initialization vector.</value>
+        public byte[] IV { get => _iv; set => _iv = value; }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Aes"/> class.
         /// </summary>
-        /// <remarks>
-        /// This implementation of the AES is specific to .NET nanoFramework.
-        /// </remarks>
         public Aes(CipherMode mode)
         {
             Mode = mode;
+
+            if (Mode == CipherMode.CBC)
+            {
+                _iv = new byte[16];
+
+                
+            }
+        }
+
+        /// <summary>
+        /// Generate a random initialization vector
+        /// </summary>
+        public void GenerateIV()
+        {
+            // Fill the IV with random data
+            Random random = new Random();
```

Comment:
@Feiko this requires the user to specifically call `GenerateIV()`...
I get it that's following up with the full .NET, but maybe we can "just" have it called in the constructor as it was before, in case the user is not aware of it or is not using it the full .NET fashion.
Thoughts?

Reply:
@josesimoes My reasoning is that generating IV is a conscious act. Because you also need the IV to decrypt the message. With LoraWAN protocol, IV is an array of 0x00's. With 0x00's as default you can always decrypt the message everywhere. Else you end up with a message that can not be decrypted because people were unaware that a random IV was used. 
If we want to move away from the big .NET I would suggest that stetting the IV will be mandatory. This way developers are always aware of the IV used to the encryption.

Reply:
Understood. A couple of things here:

1. In full .NET IV is generated automatically with a RANDOM array. Check the sample [here](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.aes?view=net-8.0&WT.mc_id=DT-MVP-5004179)  It clearly states that is filled with random bytes AND that happens in the call to `Create()`. Therefore my take on having it done in the constructor as it was.

2. Got your point on the way IV is used in LoraWAN. But, in order to do that, a developer has to set the IV property to an all 0s array. Which, actually is already done if you don't call the `GenerateIV()` as that is the default for byte[].

3. This ends up being our take on this: if we follow .NET we have to generate that random array, if we want to break, we can have that called separatly. Anyway this can be easily checked by the developers as the IV property is readable. I think we should follow the .NET behavior for the same reasons we always do: code compatibility, examples found on MS Learn, web, stackoverflow, etc. For embedded developers (for which we're paving the ground) and in particular for the ones dealing with LoraWAN (which are be a sub-set of the library users) let's make sure to add note on the readme and samples, making it very clear that they have to set IV to all 0s before calling the Encrypt/Decrypt.

Reply:
+1 with José to follow the .NET behavior which doesn't seems to prevent using the lib. Documentation and sample should be enough.

Reply:
@josesimoes 

> Understood. A couple of things here:
> 
> 1. In full .NET IV is generated automatically with a RANDOM array. Check the sample [here](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.aes?view=net-8.0&WT.mc_id=DT-MVP-5004179)  It clearly states that is filled with random bytes AND that happens in the call to `Create()`. Therefore my take on having it done in the constructor as it was.
> 2. Got your point on the way IV is used in LoraWAN. But, in order to do that, a developer has to set the IV property to an all 0s array. Which, actually is already done if you don't call the `GenerateIV()` as that is the default for byte[].
> 3. This ends up being our take on this: if we follow .NET we have to generate that random array, if we want to break, we can have that called separatly. Anyway this can be easily checked by the developers as the IV property is readable. I think we should follow the .NET behavior for the same reasons we always do: code compatibility, examples found on MS Learn, web, stackoverflow, etc. For embedded developers (for which we're paving the ground) and in particular for the ones dealing with LoraWAN (which are be a sub-set of the library users) let's make sure to add note on the readme and samples, making it very clear that they have to set IV to all 0s before calling the Encrypt/Decrypt.

Agreed indeed, we should follow the big .NET on this. I was unaware that I was deviating from the big .NET, sorry for the confusion.